### ISSUE TITLE
Screenshots appear in alphabetical order of their captions

### ISSUE THREAD

**Issue by cassidyjames** [flathub-infra/website#1995]
It looks like https://github.com/flathub/linux-store-frontend/issues/80 is still present; I would expect them to show in the order defined in the AppStream data instead.

**Comment by razzeee** [comment]
Please post an actual example

**Comment by starnight** [comment]
We have some captured pictures in the ticket [Key on EOS: App center screenshots have wrong order](https://github.com/endlessm/endless-key-flatpak/issues/50).
![image](https://github.com/flathub/website/assets/2100946/0d08cd16-f5bd-4cc5-986c-4d9e7d0c477b)
The order should be:
![image](https://github.com/flathub/website/assets/2100946/0122e740-ab66-435a-b5ca-116b85f9a43b)

The screenshots appear in alphabetical order of their captions.  Then, we [Add numbers to the screenshot captions](https://github.com/endlessm/endless-key-flatpak/commit/4a22e99ceecc6c34624b9df269df883b38a4ea5b).

**Comment by razzeee** [comment]
Please don't add numbers

**Comment by cassidyjames** [comment]
So yeah for context, we had this:

https://github.com/endlessm/endless-key-flatpak/blob/v0.3/data/metainfo/org.endlessos.Key.metainfo.xml.in.in#L39-L64

We noticed in the Flathub web front-end (in addition to GNOME Software, though I understand that's a separate issue) that the screenshots were out of order compared to how they are listed in the MetaInfo.

After researching a bit, I found https://github.com/flathub/linux-store-frontend/issues/80, which when I compared to our listing, showed that Flathub was indeed putting the screenshots in alphabetical order of the captions.

As a workaround/to test, we added numbers to the captions and sure enough, Flathub (and GNOME Software) presented them in alphabetical (well, numerical) order. I fully understand that’s not a good practice and should not be the solution here; we can revert that if we can find an acceptable alternative.

https://github.com/endlessm/endless-key-flatpak/blob/v0.5/data/metainfo/org.endlessos.Key.metainfo.xml.in.in#L39-L64

My suspicion is that the library that is parsing the MetaInfo is somehow reordering the nodes in alphabetical order. While the AppStream spec doesn't appear to make any guarantees regarding the order of screenshots, it seems strange to disregard the order app developers chose—and alphabetical based on caption feels actively worse than retaining the original order from the MetaInfo.

This is compounded by the fact that we see the same behavior in GNOME Software; perhaps it has the same behavior because it is using the same AppStream library? If so, we could definitely move this issue upstream.
